components/Msig/CreateConfirm.tsx

```diff
@@ -0,0 +1,201 @@
+import { useEffect, useState } from 'react'
+import { useRouter } from 'next/router'
+import LotusRPCEngine from '@glif/filecoin-rpc-client'
+import {
+  IconPending,
+  Dialog,
+  ShadowBox,
+  ButtonRowCenter,
+  ButtonV2,
+  SmartLink,
+  ErrorBox,
+  WarningBox,
+  navigate,
+  useWallet,
+  useMessageQuery
+} from '@glif/react-components'
+
+import { useMsig } from '../../MsigProvider'
+import { PAGE } from '../../constants'
+import getAddrFromReceipt from '../../utils/getAddrFromReceipt'
+
+interface Receipt {
+  ExitCode: number
+  GasUsed: number
+  Return: string
+}
+
+export const CreateConfirm = () => {
+  const router = useRouter()
+  const { cid } = router.query
+  const { setMsigActor, Address } = useMsig()
+  const wallet = useWallet()
+
+  // Confirmation state
+  const [receipt, setReceipt] = useState<Receipt | null>(null)
+  const [receiptError, setReceiptError] = useState<boolean>(false)
+  const [creationError, setCreationError] = useState<boolean>(false)
+
+  // Get environment variables
+  const { NEXT_PUBLIC_EXPLORER_URL: explorerUrl } = process.env
+  const { NEXT_PUBLIC_LOTUS_NODE_JSONRPC: apiAddress } = process.env
+
+  // Get the message by cid
+  const { data: messageData, error: messageError } = useMessageQuery({
+    variables: { cid: cid as string },
+    pollInterval: 500,
+    skip: !cid
+  })
+
+  // Get the receipt after receiving the message
+  useEffect(() => {
+    if (messageData?.message) {
+      setTimeout(() => {
+        const lCli = new LotusRPCEngine({ apiAddress })
+        lCli
+          .request<Receipt>('StateGetReceipt', { '/': cid }, null)
+          .then((receipt) => setReceipt(receipt))
+          .catch(() => setReceiptError(true))
+      }, 10000)
+    }
+  }, [messageData?.message, apiAddress, cid])
+
+  // Get the safe address after receiving the receipt
+  useEffect(() => {
+    if (receipt) {
+      try {
+        // Verify the exit code
+        if (receipt.ExitCode !== 0) throw new Error('Safe creation failed')
+
+        // Verify the safe address
+        const { robust } = getAddrFromReceipt(receipt.Return)
+        if (!robust) throw new Error('Failed to get address from receipt')
+
+        // Set the safe address
+        setMsigActor(robust)
+      } catch (e) {
+        setCreationError(true)
+      }
+    }
+  }, [receipt, setMsigActor])
```

Comment:
Can we put this in the same `useEffect` as above if we get a receipt? Feels like we can get rid of one of them..

Reply:
Fixed in https://github.com/glifio/safe/pull/107/files